### PR TITLE
From "term" to "tag" - We are now using "tag" where appropriate

### PR DESCRIPTION
**Pull request by annando**
This PR doesn't cover all steps in the conversion process from the "term" table to the "tag" table. But it is getting large and many things had been done here.

There is now a post update function that fills the "tag" table with content from previous posts. Also we are using the new tables and constants at all appropriate places like the search and the fetching of hashtags.

The next PR will then remove all old parts that now aren't used anymore.

### CODE REVIEW

File: src/Model/Tag.php [hunk 1]
```diff
@@ -282,6 +306,114 @@ private static function getTypeForHash(string $hash)
...
+
+	/**
+	 * Retrieves the terms from the provided type(s) associated with the provided item ID.
+	 *
+	 * @param int       $item_id
+	 * @param int|array $type
+	 * @return array
+	 * @throws \Exception
+	 */
+	public static function ArrayFromURIId(int $uri_id, array $type = [self::HASHTAG, self::MENTION, self::IMPLICIT_MENTION, self::EXCLUSIVE_MENTION])
```
Comment (MrPetovan) [inline src/Model/Tag.php:319]: Please use `camelCase` for function names.
```suggestion
	public static function getByURIId(int $uri_id, array $type = [self::HASHTAG, self::MENTION, self::IMPLICIT_MENTION, self::EXCLUSIVE_MENTION])
```

File: src/Model/Tag.php [hunk 2]
```diff
@@ -282,6 +306,114 @@ private static function getTypeForHash(string $hash)
...
+	/**
+	 * Search posts for given tag
+	 *
+	 * @param string $search
+	 * @param integer $uid
+	 * @param integer $start
+	 * @param integer $limit
+	 * @return array with URI-ID
+	 */
+	public static function getURIIdListForTag(string $search, int $uid = 0, int $start = 0, int $limit = 100)
```
[the remark this answers is not in the document]
Comment (annando) [reply src/Model/Tag.php:400]: It returns the URI-ID of the given tags - and it works with the tag-tables. So from my point of view it belongs to the tag-class.

File: src/Model/Term.php [hunk 3]
```diff
@@ -176,7 +86,7 @@ public static function tagTextFromItemId($item_id)
 	 * @return array
 	 * @throws \Exception
 	 */
-	public static function tagArrayFromItemId($item_id, $type = [self::HASHTAG, self::MENTION])
+	private static function tagArrayFromItemId($item_id, $type = [self::HASHTAG, self::MENTION])
```
Comment (MrPetovan) [inline src/Model/Term.php:89]: Please rename this method to `getByItemId`
```suggestion
	private static function getByItemId($item_id, $type = [self::HASHTAG, self::MENTION])
```

Comment (annando) [reply src/Model/Term.php:89]: This function will be removed in one of the upcoming PRs anyhow :-)

**Comment by MrPetovan** [comment]
Are you done? I see you still are adding changes.

**Comment by annando** [comment]
I'm done now.

**Comment by AlfredSK** [comment]
Please have a look at the fatal error https://github.com/friendica/friendica/issues/8473#issuecomment-622605541 and the notice https://github.com/friendica/friendica/issues/8475#issuecomment-622606137
They always occur together (with identical timestamps) after I pulled this PR.

**Comment by annando** [comment]
@AlfredSK PR is done.

**Pull request merged**